tests/cephfs/cephfs_vol_management/cephfs_vol_mgmt_data_migrate.py

```diff
@@ -0,0 +1,169 @@
+        fs_util.auth_list(clients)
+        default_fs = "cephfs"
+        if build.startswith("4"):
+            # create EC pool
+            list_cmds = [
+                "ceph fs flag set enable_multiple true",
+                "ceph osd pool create cephfs-data-ec 64 erasure",
+                "ceph osd pool create cephfs-metadata 64",
+                "ceph osd pool set cephfs-data-ec allow_ec_overwrites true",
+                "ceph fs new cephfs-ec cephfs-metadata cephfs-data-ec --force",
+            ]
+            if fs_util.get_fs_info(clients[0], "cephfs_new"):
+                default_fs = "cephfs_new"
+                list_cmds.append("ceph fs volume create cephfs")
+            for cmd in list_cmds:
+                clients[0].exec_command(sudo=True, cmd=cmd)
+
+        log.info("Create SubVolumeGroups on each filesystem")
+        subvolumegroup_list = [
+            {"vol_name": default_fs, "group_name": "subvolgroup_1"},
+            {"vol_name": "cephfs-ec", "group_name": "subvolgroup_ec1"},
+        ]
+        for subvolumegroup in subvolumegroup_list:
+            fs_util.create_subvolumegroup(clients[0], **subvolumegroup)
+
+        log.info("Create 2 Sub volumes on each of the pool with Size 5GB")
+        subvolume_list = [
+            {
+                "vol_name": default_fs,
+                "subvol_name": "subvol_1",
+                "group_name": "subvolgroup_1",
+                "size": "5368709120",
+            },
+            {
+                "vol_name": "cephfs-ec",
+                "subvol_name": "subvol_2",
+                "group_name": "subvolgroup_ec1",
+                "size": "5368709120",
+            },
+        ]
+        for subvolume in subvolume_list:
+            fs_util.create_subvolume(clients[0], **subvolume)
+
+        mounting_dir = "".join(
+            random.choice(string.ascii_lowercase + string.digits)
+            for _ in list(range(10))
+        )
+        log.info("Mount 1 subvolume on kernel and 1 subvloume on Fuse → Client1")
+        kernel_mounting_dir_1 = f"/mnt/cephfs_kernel{mounting_dir}_1/"
+        mon_node_ips = fs_util.get_mon_node_ips()
+        log.info("Get the path of sub volume")
+        subvol_path, rc = clients[0].exec_command(
+            sudo=True,
+            cmd=f"ceph fs subvolume getpath {default_fs} subvol_1 subvolgroup_1",
+        )
+        fs_util.kernel_mount(
+            [clients[0]],
+            kernel_mounting_dir_1,
+            ",".join(mon_node_ips),
+            sub_dir=f"{subvol_path.read().decode().strip()}",
+        )
+
+        subvol_path, rc = clients[0].exec_command(
+            sudo=True,
+            cmd="ceph fs subvolume getpath cephfs-ec subvol_2 subvolgroup_ec1",
+        )
+        fuse_mounting_dir_1 = f"/mnt/cephfs_fuse{mounting_dir}_EC_1/"
+        fs_util.fuse_mount(
+            [clients[0]],
+            fuse_mounting_dir_1,
+            extra_params=f" -r {subvol_path.read().decode().strip()} --client_fs cephfs-ec",
```

Comment:
Hi Hemanth
I think this will not work in 4.x builds.
Can you review this once

Reply:
Updated the script wirh if condition to run on both 4.x and 5.x. Thanks Amar
Latest logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-ZHYT8K/